From bf346fec5fd5e50d4b76cbd81063d17de31a7f19 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Wed, 13 Jan 2016 17:31:20 +0800
Subject: [PATCH 24/39] package-kernel: don't allow building when tree is dirty

When there are local changes, the package won't accurately reflect what's
in the repo.  This is a bad thing.  It also leads to version numbers with
the string -dirty appended.

If the repo has local changes, refuse to build.

Signed-off-by: Sean Cross <xobs@kosagi.com>

(sakaki: context modified for 4.7.2)
(GuzTech: context modified for 4.13-rc6)
---
 package-kernel.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/package-kernel.sh b/package-kernel.sh
index e303f087cb8f..9b8fd1929747 100755
--- a/package-kernel.sh
+++ b/package-kernel.sh
@@ -3,6 +3,12 @@
 threads=$(($(grep '^processor' /proc/cpuinfo  | cut -d: -f2 | tail -n1)+1))
 version=4.13-rc6-novena
 
+if git diff-index --name-only HEAD | grep -qv "^scripts/package"; then
+	echo "Repo has local changes.  Stash or commit them."
+	exit 1
+fi
+
+
 echo "Building v${version} with ${threads} threads"
 git tag -d "v${version}" 2> /dev/null
 git tag "v${version}"
-- 
2.14.1

